From ec29304e4a0b2e9ffccc935529c55fe78d3e18ea Mon Sep 17 00:00:00 2001
From: Mingcong Bai <baimingcong@uniontech.com>
Date: Fri, 28 Jun 2024 15:11:14 +0800
Subject: [PATCH 091/360] irqchip: drop duplicate __maybe_unused gic_read_iar()
 in irq-gic-phytium-2500.c

This was already defined in arch/arm64/include/asm/arch_gicv3.h.

Signed-off-by: Mingcong Bai <baimingcong@uniontech.com>
---
 drivers/irqchip/irq-gic-phytium-2500.c | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/drivers/irqchip/irq-gic-phytium-2500.c b/drivers/irqchip/irq-gic-phytium-2500.c
index b4a749b87383..1488e7bae79f 100644
--- a/drivers/irqchip/irq-gic-phytium-2500.c
+++ b/drivers/irqchip/irq-gic-phytium-2500.c
@@ -243,17 +243,6 @@ static void gic_redist_wait_for_rwp(void)
 	gic_do_wait_for_rwp(gic_data_rdist_rd_base(), GICR_CTLR_RWP);
 }
 
-#ifdef CONFIG_ARM64
-
-static u64 __maybe_unused gic_read_iar(void)
-{
-	if (cpus_have_const_cap(ARM64_WORKAROUND_CAVIUM_23154))
-		return gic_read_iar_cavium_thunderx();
-	else
-		return gic_read_iar_common();
-}
-#endif
-
 static void gic_enable_redist(bool enable)
 {
 	void __iomem *rbase;
-- 
2.20.1

